src/fastoad/openmdao/_utils.py

```diff
@@ -16,52 +16,57 @@
 
 from contextlib import contextmanager
 from copy import deepcopy
-from typing import List, Tuple
+from typing import List, Tuple, TypeVar
 
 import numpy as np
 import openmdao.api as om
 from deprecated import deprecated
+from openmdao.core.constants import _SetupStatus
 from openmdao.utils.mpi import FakeComm
 
+T = TypeVar("T", bound=om.Problem)
 
-@contextmanager
-def problem_without_mpi(problem: om.Problem) -> om.Problem:
-    """
-    Context manager that delivers a copy of the given OpenMDAO problem.
 
-    A deepcopy operation may crash if problem.comm is not pickle-able, like a
+def get_problem_copy_without_mpi(problem: T) -> T:
```

Comment:
I'm not sure about the name of the function. From the description, I seem to understand that this function allows to avoid the problem with MPI by removing the non-pickable object beforehand and readding it afterwards, which would work regardless of whether the problem has MPI or not. So the `wihout_mpi` seems more like a `mpi_safe` ?

Reply:
Good point. Name adopted in 52fe6497df9f02e88896d683a8b1bf954fd1dd21